
package Common;

option optimize_for = LITE_RUNTIME;

message RequestGameInstanceList
{
    required string game_name               = 1;
}

message GameInstanceClient
{
    required uint32 index                   = 1;
    required string desc                    = 2;    // in Texas, desc is limit_min,limit_max,default_carry,max_player_count,sb,bb,item_price,type
    required uint32 cur_player_count        = 3;
    required uint32 max_player_count        = 4;
}
message GameInstanceListClient
{
    required string game_name               = 1;
    repeated GameInstanceClient instances   = 2;
}

message GameInstanceServer
{
    required string desc                    = 1;
    required uint32 instance_count          = 2;
    required uint32 max_player_count        = 3;
}
message GameInstanceListServer
{
    required string game_name               = 1;
    repeated GameInstanceServer instances   = 2;
}
